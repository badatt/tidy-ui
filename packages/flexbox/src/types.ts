import { HTMLAttributes } from 'react';
import { IBaseProps, TCreateArrayWithLengthX, TNumericRange } from '@tidy-ui/types';

/**
 * css `flex-direction`
 */
export enum FlexDirection {
  'row',
  'row-reverse',
  'column',
  'column-reverse',
}

/**
 * `flex-direction` props
 */
export type TFlexDirection = keyof typeof FlexDirection;

/**
 * css `align-content`
 */
export enum AlignContent {
  'normal',
  'baseline',
  'first baseline',
  'last baseline',
  'space-between',
  'space-around',
  'space-evenly',
  'stretch',
  'unsafe',
  'safe',
  'center',
  'start',
  'end',
  'flex-start',
  'flex-end',
}

/**
 * `align-content` props
 */
export type TAlignContent = keyof typeof AlignContent;

/**
 * css `align-items`
 */
export enum AlignItems {
  'normal',
  'stretch',
  'baseline',
  'first baseline',
  'last baseline',
  'safe',
  'unsafe',
  'flex-start',
  'flex-end',
  'center',
  'start',
  'end',
  'self-start',
  'self-end',
}

/**
 * `align-items` props
 */
export type TAlignItems = keyof typeof AlignItems;

/**
 * css `align-self`
 */
export enum AlignSelf {
  'auto',
  'normal',
  'stretch',
  'baseline',
  'first baseline',
  'last baseline',
  'unsafe',
  'safe',
  'center',
  'start',
  'end',
  'self-start',
  'self-end',
  'flex-start',
  'flex-end',
}

/**
 * `align-self` props
 */
export type TAlignSelf = keyof typeof AlignSelf;

/**
 * css `justify-content`
 */
export enum JustifyContent {
  'center',
  'start',
  'end',
  'flex-start',
  'flex-end',
  'left',
  'right',
  'normal',
  'space-between',
  'space-around',
  'space-evenly',
  'stretch',
  'safe',
  'unsafe',
}
/**
 * `justify-content` props
 */
export type TJustifyContent = keyof typeof JustifyContent;

/**
 * FlexBox props
 */
export interface IFlexBoxProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * css `align-content` equivalent
   *
   * @default 'flex-start'
   */
  alc?: TAlignContent;

  /**
   * css `align-items` equivalent
   */
  ali?: TAlignItems;

  /**
   * css `column-gap` equivalent
   */
  cgp?: string;

  /**
   * If `true`, places the items inside flexbox center both on main and cross axis.
   * Set `align-content` & `justify-content` to center. Set to full height if height is not
   * provided
   *
   * @default false
   */
  ctr?: boolean;

  /**
   * css `flex-direction` equivalent
   *
   * @default 'row'
   */
  fld?: TFlexDirection;

  /**
   * css `flow` equivalent
   */
  flow?: string;

  /**
   * If `true`, set the FlexBox full height
   *
   * @default false
   */
  fuh?: boolean;

  /**
   * css `gap` equivalent
   */
  gap?: string;

  /**
   * css `justify-content` equivalent
   *
   * @default 'flex-start'
   */
  jsc?: TJustifyContent;

  /**
   * If `true`, sets `flex-wrap: nowrap` the items
   *
   * @default false
   */
  nowrap?: boolean;

  /**
   * css `row-gap` equivalent
   */
  rgp?: string;
}

/**
 * FlexItem props
 */
export interface IFlexItemProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * css `align-self` equivalent
   */
  als?: TAlignSelf;

  /**
   * css `flex-basis` equivalent
   */
  fbs?: string;

  /**
   * css `flex-grow` equivalent
   */
  fgo?: number;

  /**
   * css `flex` equivalent
   */
  flx?: string;

  /**
   * css `flex-shrink` equivalent
   */
  fsk?: number;

  /**
   * If `true`, set the FlexItem full width
   *
   * @default false
   */
  fuw?: boolean;

  /**
   * css `order` equivalent
   */
  ord?: number;

  /**
   * spanning inside a FlexBox, 24 spans to 100% width, 1 spans to 1/24th percent width
   */
  span?: TNumericRange<TCreateArrayWithLengthX<1>, 24>;
}
