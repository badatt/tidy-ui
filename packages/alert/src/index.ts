export * from './Alert';
export * from './AlertFooter';
export * from './AlertTitle';
