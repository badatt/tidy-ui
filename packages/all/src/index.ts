export * from '@tidy-ui/commons';
export * from '@tidy-ui/feedback';
export * from '@tidy-ui/form';
export * from '@tidy-ui/layout';
export * from '@tidy-ui/navigation';
export * from '@tidy-ui/presentation';
