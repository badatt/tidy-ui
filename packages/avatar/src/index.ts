export * from './Avatar';
export * from './AvatarBadge';
export * from './AvatarGroup';
