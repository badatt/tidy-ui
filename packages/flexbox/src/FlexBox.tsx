import React, { forwardRef, useEffect, useState } from 'react';
import { css, styled } from '@tidy-ui/commons';
import { IFlexBoxProps } from './types';

const FlexBoxRoot = styled.div<IFlexBoxProps>`
  ${({ h, w, fuh, alc, ali, cgp, fld, flow, gap, jsc, rgp, fwp }) => css`
    display: flex;
    flex-direction: ${fld};
    justify-content: ${jsc};
    flex-wrap: ${fwp};
    ${ali &&
    css`
      align-items: ${ali};
    `}
    ${!ali &&
    alc &&
    css`
      align-content: ${alc};
    `}
    
    ${w &&
    css`
      width: ${w};
    `}
    ${!fuh &&
    h &&
    css`
      height: ${h};
    `}
    ${fuh &&
    css`
      height: 100%;
    `}
    ${gap &&
    css`
      gap: ${gap};
    `}
    ${cgp &&
    css`
      column-gap: ${cgp};
    `}
    ${rgp &&
    css`
      row-gap: ${rgp};
    `}
    ${flow &&
    css`
      flex-flow: ${flow};
    `}
  `}
`;

const FlexBox = forwardRef<HTMLDivElement, IFlexBoxProps>((props, ref) => {
  const { children, ...r } = props;
  const [rest, setRest] = useState<IFlexBoxProps>(r);

  useEffect(() => {
    if (r.ctr) {
      setRest({
        ...rest,
        alc: 'center',
        fuh: rest.h ? false : true,
        jsc: 'center',
      });
    }
  }, [r.ctr]);

  return (
    <FlexBoxRoot ref={ref} role="presentation" {...rest}>
      {children}
    </FlexBoxRoot>
  );
});

FlexBox.defaultProps = {
  alc: 'flex-start',
  ctr: false,
  fld: 'row',
  fuh: false,
  fwp: 'wrap',
  jsc: 'flex-start',
};

export { FlexBox };
