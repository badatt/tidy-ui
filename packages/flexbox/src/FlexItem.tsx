import React, { forwardRef } from 'react';
import { css, styled } from '@tidy-ui/commons';
import { IFlexItemProps } from './types';

const FlexItemRoot = styled.div<IFlexItemProps>`
  ${({ als, fbs, fgo, flx, fsk, ord }) => css`
    ${als &&
    css`
      align-self: ${als};
    `}
    ${fbs &&
    css`
      flex-basis: ${fbs};
    `}
    ${fgo &&
    css`
      flex-grow: ${fgo};
    `}
    ${flx &&
    css`
      flex: ${flx};
    `}
    ${fsk &&
    css`
      flex-shrink: ${fsk};
    `}
    ${ord &&
    css`
      order: ${ord};
    `}
  `}
`;

const FlexItem = forwardRef<HTMLDivElement, IFlexItemProps>((props, ref) => {
  const { children, ...rest } = props;
  return (
    <FlexItemRoot ref={ref} role="presentation" {...rest}>
      {children}
    </FlexItemRoot>
  );
});

FlexItem.defaultProps = {};

export { FlexItem };
