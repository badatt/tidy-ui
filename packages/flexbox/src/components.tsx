import { applyStandardOverrideStyles, css, styled } from '@tidy-ui/commons';
import { IFlexBoxProps, IFlexItemProps } from './types';

const FlexBoxRoot = styled.div<IFlexBoxProps>`
  ${({ fuh, alc, ali, cgp, fld, flow, gap, jsc, rgp, nowrap }) => css`
    display: flex;
    flex-direction: ${fld};
    justify-content: ${jsc};
    flex-wrap: ${nowrap ? 'nowrap' : 'wrap'};
    ${ali &&
    css`
      align-items: ${ali};
    `}
    ${!ali &&
    alc &&
    css`
      align-content: ${alc};
    `}
    ${fuh &&
    css`
      height: 100%;
    `}
    ${gap &&
    css`
      gap: ${gap};
    `}
    ${cgp &&
    css`
      column-gap: ${cgp};
    `}
    ${rgp &&
    css`
      row-gap: ${rgp};
    `}
    ${flow &&
    css`
      flex-flow: ${flow};
    `}
  `}
  ${applyStandardOverrideStyles}
`;

const FlexItemRoot = styled.div<IFlexItemProps>`
  position: relative;
  display: block;
  min-height: 1px;
  ${({ als, fbs, fgo, flx, fsk, fuw, ord, span }) => css`
    ${als &&
    css`
      align-self: ${als};
    `}
    ${fbs &&
    css`
      flex-basis: ${fbs};
    `}
    ${fgo &&
    css`
      flex-grow: ${fgo};
    `}
    ${flx &&
    css`
      flex: ${flx};
    `}
    ${fsk &&
    css`
      flex-shrink: ${fsk};
    `}
    ${ord &&
    css`
      order: ${ord};
    `}
    ${span &&
    span > 0 &&
    !fuw &&
    css`
      width: ${() => (span / 24) * 100}%;
    `}
    ${fuw &&
    css`
      width: 100%;
    `}
  `}
  ${applyStandardOverrideStyles}
`;

export { FlexBoxRoot, FlexItemRoot };
