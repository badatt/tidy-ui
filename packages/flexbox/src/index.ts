export * from './FlexBox';
export * from './FlexItem';
