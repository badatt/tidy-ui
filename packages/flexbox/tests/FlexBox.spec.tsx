/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidLight } from '@tidy-ui/commons';
import { FlexBox } from '../src';
import {
  AlignContent,
  AlignItems,
  FlexDirection,
  IFlexBoxProps,
  JustifyContent,
  TAlignContent,
  TAlignItems,
  TFlexDirection,
  TJustifyContent,
} from '../src/types';

const height = '350px';

function FlexBoxWrapper(props: IFlexBoxProps) {
  return (
    <FlexBox {...props}>
      <div>1</div>
      <div>2</div>
      <div>3</div>
      <div>4</div>
      <div>5</div>
      <div>6</div>
      <div>7</div>
      <div>8</div>
    </FlexBox>
  );
}

describe('FlexBox', () => {
  it('Basic render', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper h={height} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Direction', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        {Object.keys(FlexDirection)
          .filter((i) => !isNaN(Number(i)))
          .map((v) => (
            <FlexBoxWrapper h={height} fld={FlexDirection[v]} key={v} />
          ))}
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Wrap', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        {[true, false].map((v, i) => (
          <FlexBoxWrapper h={height} nowrap={v} key={i} />
        ))}
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Centered', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper ctr />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Centered with height set', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper ctr h={height} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Full height', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper fuh />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Gap', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper gap="8px" h={height} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Column gap', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper cgp="8px" h={height} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Row gap', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper rgp="8px" h={height} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Align content', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        {Object.keys(AlignContent)
          .filter((i) => !isNaN(Number(i)))
          .map((v) => (
            <FlexBoxWrapper h={height} alc={AlignContent[v]} key={v} />
          ))}
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Align items', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        {Object.keys(AlignItems)
          .filter((i) => !isNaN(Number(i)))
          .map((v) => (
            <FlexBoxWrapper h={height} ali={AlignItems[v]} key={v} />
          ))}
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Justify content', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        {Object.keys(JustifyContent)
          .filter((i) => !isNaN(Number(i)))
          .map((v) => (
            <FlexBoxWrapper h={height} jsc={JustifyContent[v]} key={v} />
          ))}
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Width', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper w="600px" />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Flow', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper flow="column wrap" />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Margin', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <FlexBoxWrapper flow="column wrap" margin="0 0 3rem 0" />
        <FlexBoxWrapper flow="column wrap" />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
