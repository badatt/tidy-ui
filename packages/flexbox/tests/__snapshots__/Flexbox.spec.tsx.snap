// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flexbox Basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

<body>
    <div>
      <div
        class="c0"
        role="presentation"
      >
        basic
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

<div>
    <div
      class="c0"
      role="presentation"
    >
      basic
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Flexbox Dark mode basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

<body>
    <div>
      <div
        class="c0"
        role="presentation"
      >
        basic
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

<div>
    <div
      class="c0"
      role="presentation"
    >
      basic
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
